packages/pwa-kit-react-sdk/src/utils/routes.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react'
+
+/**
+ * Transforms a URL mapping from the Shopper Search getUrlMapping API to a routes config.
+ * https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-seo?meta=getUrlMapping
+ *
+ * @param {Object} urlMapping - The URL mapping object returned from the Shopper SEO getURLMapping API.
+ * @param {Object} component - The component to be rendered for the route.
+ */
+export const transformUrlMappingToRoute = (path, urlMapping, component) => {
+    let Component, props
+
+    // Resource type is not defined for redirects with a URL destination
+    const isRedirect = !urlMapping.resourceType
+
+    if (isRedirect) {
+        Component = Redirect
+        props = {
+            to: urlMapping.destinationUrl
+        }
+    } else {
+        // TODO: support "content_asset" resource types
+        Component = component
+        props = {
+            [`${urlMapping.resourceType}Id`]: urlMapping.resourceId
+        }
```

Comment:
@bendvc still need to add support for mapping `content_asset` resource types to a component